### ISSUE TITLE
Connection leak when running H2 tests

### ISSUE THREAD

**Issue by pierre** [killbill/killbill#305]
In the `kares-performance` branch, H2 server tests don't pass anymore, most likely because of a connection leak (bumping the max number of active connections from 10 to 30 doesn't help).

```
org.skife.jdbi.v2.exceptions.UnableToObtainConnectionException: java.sql.SQLException: Login timeout
	at org.skife.jdbi.v2.DBI.open(DBI.java:228)
	at org.skife.jdbi.v2.sqlobject.OnDemandHandleDing.getHandle(OnDemandHandleDing.java:38)
	at org.skife.jdbi.v2.sqlobject.OnDemandHandleDing.retain(OnDemandHandleDing.java:46)
	at org.skife.jdbi.v2.sqlobject.SqlObject.invoke(SqlObject.java:174)
	at org.skife.jdbi.v2.sqlobject.SqlObject$2.intercept(SqlObject.java:92)
	at org.skife.jdbi.v2.sqlobject.CloseInternalDoNotUseThisClass$$EnhancerByCGLIB$$44022973.getReadyEntries(<generated>)
	at org.killbill.queue.DBBackedQueue.fetchReadyEntries(DBBackedQueue.java:539)
	at org.killbill.queue.DBBackedQueue.getReadyEntries(DBBackedQueue.java:277)
	at org.killbill.bus.DefaultPersistentBus.doProcessEvents(DefaultPersistentBus.java:119)
	at org.killbill.queue.DefaultQueueLifecycle$1.run(DefaultQueueLifecycle.java:111)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.sql.SQLException: Login timeout
	at org.h2.jdbcx.JdbcConnectionPool.getConnection(JdbcConnectionPool.java:208)
	at net.sf.log4jdbc.sql.jdbcapi.DataSourceSpy.getConnection(DataSourceSpy.java:147)
	at org.skife.jdbi.v2.DataSourceConnectionFactory.openConnection(DataSourceConnectionFactory.java:36)
	at org.skife.jdbi.v2.DBI.open(DBI.java:210)
	... 12 common frames omitted
```

**Comment by pierre** [comment]
This is a regression in jdbi, see https://github.com/jdbi/jdbi/pull/151.

**Comment by pierre** [comment]
https://github.com/jdbi/jdbi/issues/151 has been merged but we cannot use the 2.60 build:

```
java.lang.NoClassDefFoundError: org/skife/jdbi/cglib/proxy/Callback
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at org.apache.maven.surefire.booter.IsolatedClassLoader.loadClass(IsolatedClassLoader.java:97)
	at org.skife.jdbi.v2.sqlobject.SqlObjectBuilder.onDemand(SqlObjectBuilder.java:66)
	at org.skife.jdbi.v2.DBI.onDemand(DBI.java:350)
```

See https://github.com/jdbi/jdbi/issues/152.

**Comment by pierre** [comment]
Fixed by https://github.com/killbill/killbill-oss-parent/commit/a72c9d6e62bd81558e6b54ff7bec04ba34fedcec.
